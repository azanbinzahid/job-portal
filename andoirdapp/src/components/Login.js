import React from 'react';
import {connect} from 'react-redux';
import {useFormik} from 'formik';
import * as yup from 'yup';
import {fetchUser, logUserOut} from '../redux/actions';
import {View, TextInput, Button, Text} from 'react-native';

const validationSchema = yup.object().shape({
  username: yup.string().max(16).required(),
  password: yup.string().required(),
});

const Login = (props) => {
  const {handleSubmit, handleChange, handleBlur, values, errors} = useFormik({
    initialValues: {
      username: '',
      password: '',
    },
    // validationSchema,
    onSubmit(values) {
      props.fetchUser(values);
    },
  });

  if (props.isLogged) {
    return (
      <View>
        <Text>You are Logged In</Text>
        <Button
          title="Go to Home"
          onPress={() => props.navigation.navigate('Home')}
        />
        <Button title="Log Out" onPress={() => props.logUserOut()} />
      </View>
    );
  }

  return (
    <View>
      <Text>Login Page</Text>
      <View>
        <TextInput
          placeholder="Username"
          value={values.username}
          onChangeText={handleChange('username')}
          onBlurText={handleBlur('username')}
        />
        {/* {errors.username} */}
        <TextInput
          placeholder="Password"
          value={values.password}
          onChangeText={handleChange('password')}
          onBlurText={handleBlur('password')}
        />
        {/* {errors.password} */}
        <Button onPress={handleSubmit} title="Login" />
      </View>
    </View>
  );
};

const mapStateToProps = (state) => ({
  isLogged: state.userReducer.loggedIn,
});

const mapDispatchToProps = (dispatch) => {
  return {
    fetchUser: (userInfo) => dispatch(fetchUser(userInfo)),
    logUserOut: () => dispatch(logUserOut()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
