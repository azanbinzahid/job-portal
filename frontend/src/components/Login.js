import React from 'react'
import {connect} from 'react-redux'
import {fetchUser} from 'redux/actions'
import { Redirect } from 'react-router-dom';
import { Form, Button, Jumbotron, Container } from 'react-bootstrap';

class Login extends React.Component {
    state = {
        username: "",
        password: ""
    }

    handleOnChange = (e) => {
        e.persist();
        this.setState(() => ({
            [e.target.name]: e.target.value 
        }))
    };


    onSubmit = (e) => {
        e.preventDefault()
        this.props.fetchUser(this.state)
    }


    render(){

        
        if (this.props.userReducer.loggedIn) {
            return <Redirect to="/"/>
        }



        return(
            <Container>
            <Jumbotron align="center">
                <h1> Login Page </h1>
            </Jumbotron>
            <Jumbotron>
            <Form onSubmit={this.onSubmit}>
                <Form.Group controlId="formBasicEmail">
                    <Form.Control 
                        type="text" 
                        name="username" 
                        placeholder="Username" 
                        value={this.state.username}
                        onChange={this.handleOnChange}
                    />
                </Form.Group>
                <Form.Group controlId="formBasicPassword">
                    <Form.Control 
                        type="password"
                        name="password"
                        placeholder="Password"
                        value={this.state.password}
                        onChange={this.handleOnChange}
                    />
                </Form.Group>
                <Button variant="primary" type="submit">
                    Login
                </Button>
            </Form>
            </Jumbotron>
            </Container>
        )
    }
}

const mapStateToProps = state => ({
    userReducer: state.userReducer
  });

const mapDispatchToProps = (dispatch) => {
    return {
        fetchUser: (userInfo) => dispatch(fetchUser(userInfo))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
