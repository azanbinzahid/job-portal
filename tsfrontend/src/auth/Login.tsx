import React, { FC } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { Form, Button, Jumbotron, Container } from "react-bootstrap";
import { useFormik } from "formik";
import * as yup from "yup";
import { fetchUser } from "redux/actions";
import { RootState } from "redux/reducers";
import TextField from "app/common/TextField";
import PageHeader from "app/common/PageHeader";

const validationSchema = yup.object().shape({
  username: yup.string().max(16).required(),
  password: yup.string().required(),
});

interface User {
  username: string;
  password: string;
}
interface Props {
  isLogged: boolean;
  fetchUser: (values: User) => void;
}

const Login: FC<Props> = (props) => {
  const { handleSubmit, handleChange, values, errors } = useFormik({
    initialValues: {
      username: "",
      password: "",
    },
    validationSchema,
    onSubmit(values) {
      props.fetchUser(values);
    },
  });

  if (props.isLogged) {
    return <Redirect to="/" />;
  }

  return (
    <Container>
      <PageHeader title="Login Page" />
      <Jumbotron>
        <Form
          onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)}
        >
          <TextField
            type="text"
            name="username"
            placeholder="Username"
            value={values.username}
            onChange={handleChange}
            error={errors.username}
          />
          <TextField
            type="password"
            name="password"
            placeholder="Password"
            value={values.password}
            onChange={handleChange}
            error={errors.password}
          />
          <Button variant="primary" type="submit">
            Login
          </Button>
        </Form>
      </Jumbotron>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  isLogged: state.userReducer.loggedIn,
});

const mapDispatchToProps = {
  fetchUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login as any);
