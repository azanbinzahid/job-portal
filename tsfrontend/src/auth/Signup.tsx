import React, { FC } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { Form, Button, Jumbotron, Container } from "react-bootstrap";
import { useFormik } from "formik";
import * as yup from "yup";
import { signUserUp } from "redux/actions";
import { RootState } from "redux/reducers";
import TextField from "../app/common/TextField";
import SectionHeading from "app/common/SectionHeading";

const validationSchema = yup.object().shape({
  username: yup.string().max(16).required(),
  email: yup.string().email().required(),
  firstName: yup.string().max(16).required(),
  lastName: yup.string().max(16).required(),
  password: yup.string().min(8).required(),
});

interface User {
  username: string;
  password: string;
  firstName: string;
  lastName: string;
  email: string;
}
interface Props {
  isLogged: boolean;
  signUserUp: (values: User) => void;
}

export const Signup: FC<Props> = (props) => {
  const { handleSubmit, handleChange, values, errors } = useFormik({
    initialValues: {
      username: "",
      password: "",
      email: "",
      firstName: "",
      lastName: "",
    },
    validationSchema,
    onSubmit(values) {
      props.signUserUp(values);
    },
  });

  if (props.isLogged) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <SectionHeading title="SignUp Page" />
      <Container>
        <Jumbotron>
          <Form
            onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)}
          >
            <TextField
              type="text"
              name="username"
              placeholder="Username"
              value={values.username}
              onChange={handleChange}
              error={errors.username}
            />
            <TextField
              type="password"
              name="password"
              placeholder="Password"
              value={values.password}
              onChange={handleChange}
              error={errors.password}
            />
            <TextField
              type="email"
              name="email"
              placeholder="Email"
              value={values.email}
              onChange={handleChange}
              error={errors.email}
            />
            <TextField
              type="text"
              name="firstName"
              placeholder="First Name"
              value={values.firstName}
              onChange={handleChange}
              error={errors.firstName}
            />

            <TextField
              type="text"
              name="lastName"
              placeholder="Last Name"
              value={values.lastName}
              onChange={handleChange}
              error={errors.lastName}
            />

            <Button variant="primary" type="submit">
              SignUp
            </Button>
          </Form>
        </Jumbotron>
      </Container>
    </>
  );
};

const mapStateToProps = (state: RootState) => ({
  isLogged: state.userReducer.loggedIn,
});

const mapDispatchToProps = {
  signUserUp,
};

export default connect(mapStateToProps, mapDispatchToProps)(Signup as any);
