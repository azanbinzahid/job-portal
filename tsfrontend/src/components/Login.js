import React from 'react'
import {connect} from 'react-redux'
import { Redirect } from 'react-router-dom';
import { Form, Button, Jumbotron, Container } from 'react-bootstrap';
import { useFormik } from "formik";
import * as yup from 'yup'
import {fetchUser} from 'redux/actions'

const validationSchema = yup.object().shape({   
    username: yup     
        .string()     
        .max(16)     
        .required(),   
    password: yup    
        .string()    
        .required(),
})

const Login = (props) => {
    const { handleSubmit, handleChange, values, errors} = useFormik({
        initialValues: {
            username: "",
            password: "",
        },
        validationSchema,
        onSubmit(values) {
            props.fetchUser(values)
        }
    })


    if (props.isLogged) {
        return <Redirect to="/"/>
    }


    return(
        <Container>
        <Jumbotron align="center">
            <h1> Login Page </h1>
        </Jumbotron>
        <Jumbotron>
        <Form onSubmit={handleSubmit}>
            <Form.Group controlId="formBasicEmail">
                <Form.Control 
                    type="text" 
                    name="username" 
                    placeholder="Username" 
                    value={values.username}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.username}
            <Form.Group controlId="formBasicPassword">
                <Form.Control 
                    type="password"
                    name="password"
                    placeholder="Password"
                    value={values.password}
                    onChange={handleChange}
                    />
            </Form.Group>

            <Button variant="primary" type="submit">
                Login
            </Button>
        </Form>
        </Jumbotron>
        </Container>
    )
}


const mapStateToProps = state => ({
    isLogged: state.userReducer.loggedIn
  });

const mapDispatchToProps = (dispatch) => {
    return {
        fetchUser: (userInfo) => dispatch(fetchUser(userInfo))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
