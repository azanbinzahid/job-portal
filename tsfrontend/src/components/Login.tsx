import React, { FunctionComponent } from 'react'
import {connect} from 'react-redux'
import { Redirect } from 'react-router-dom';
import { Form, Button, Jumbotron, Container } from 'react-bootstrap';
import { useFormik } from "formik";
import * as yup from 'yup'
import {fetchUser} from 'redux/actions'
import { RootState } from 'redux/reducers';

const validationSchema = yup.object().shape({   
    username: yup     
        .string()     
        .max(16)     
        .required(),   
    password: yup    
        .string()    
        .required(),
})

interface User {
    username: string,
    password: string,
}
interface Props {
    isLogged: boolean
    fetchUser: (values: User) => void
}

const Login: FunctionComponent<Props> = (props) => {
    const { handleSubmit, handleChange, values, errors} = useFormik({
        initialValues: {
            username: "",
            password: "",
        },
        validationSchema,
        onSubmit(values) {
            props.fetchUser(values)
        }
        
    })


    if (props.isLogged) {
        return <Redirect to="/"/>
    }


    return(
        <Container>
        <Jumbotron <React.ElementType> align="center">
            <h1> Login Page </h1>
        </Jumbotron>
        <Jumbotron>
        <Form onSubmit={(e: React.FormEvent<HTMLFormElement>)=>handleSubmit(e)}>
            <Form.Group controlId="formBasicEmail">
                <Form.Control 
                    type="text" 
                    name="username" 
                    placeholder="Username" 
                    value={values.username}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.username}
            <Form.Group controlId="formBasicPassword">
                <Form.Control 
                    type="password"
                    name="password"
                    placeholder="Password"
                    value={values.password}
                    onChange={handleChange}
                    />
            </Form.Group>

            <Button variant="primary" type="submit">
                Login
            </Button>
        </Form>
        </Jumbotron>
        </Container>
    )
}


const mapStateToProps = (state: RootState) => ({
    isLogged: state.userReducer.loggedIn
  });

const mapDispatchToProps = {
    fetchUser
}

export default connect(mapStateToProps, mapDispatchToProps)(Login as any);
