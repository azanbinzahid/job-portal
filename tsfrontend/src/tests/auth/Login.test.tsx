import React from "react";
import { shallow } from "enzyme";
import { Login } from "auth/Login";
test("renders the component", () => {
  const component = shallow(<Login isLogged={false} fetchUser={jest.fn()} />);
  expect(component).toMatchSnapshot();
});
test("renders the component", () => {
  const component = shallow(<Login isLogged={true} fetchUser={jest.fn()} />);
  expect(component).toMatchSnapshot();
});
