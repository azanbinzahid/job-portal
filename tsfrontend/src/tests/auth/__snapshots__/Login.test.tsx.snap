// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Container
  fluid={false}
>
  <PageHeader
    title="Login Page"
  />
  <Jumbotron
    fluid={false}
  >
    <Form
      inline={false}
      onSubmit={[Function]}
    >
      <TextField
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <TextField
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <Button
        active={false}
        disabled={false}
        type="submit"
        variant="primary"
      >
        Login
      </Button>
    </Form>
  </Jumbotron>
</Container>
`;

exports[`renders the component 2`] = `
<Redirect
  to="/"
/>
`;
